addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -253,25 +233,25 @@ def _onchange_product_id(self):
         if self.product_id:
             self.product_uom_id = self.product_id.uom_po_id
             self.product_qty = 1.0
-        if not self.schedule_date:
-            self.schedule_date = self.requisition_id.schedule_date
+            self.price_unit = 0.0
+            if self.requisition_id.requisition_type == 'purchase_template':
+                supplier_info = self.product_id.seller_ids.filtered(
+                    lambda s: s.partner_id == self.requisition_id.vendor_id
+                    )
+                self.price_unit = min(supplier_info.mapped('price')) if supplier_info else self.product_id.standard_price
```

Comment:
It worked before without having this block of code. Why do you need it now?

Reply:
Hello @amoyaux 
This block of code is for `purchase template`.
To get the `pricelist price` if available else `standard price` for `purchase_requisition_line`.
Earlier the price unit was `always 0`.
Now `price unit` is `0` for `blanket order` and for `purchase template` it will `pricelist price` if available else `standard price`

Reply:
And why doesn't it fallback on classic behavior of purchase module?

Reply:
My bad, you can use the _select_seller method and it would be great to have price_unit as a compute store instead of an onchange